ex4B/ex4.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Write a function that convers a string into number; return null if the string
+ * doesn't contain a valid number.
+ *
+ * For example:
+ * toNumber("71") === 71
+ * toNumber("a7") === null
+ */
+function toNumber(str) {
+
+	var strNum = '', strDec = '', number = null, i = 0, allowDot = true, negative = false;
+
+	// Negative sige can appear only at the start of string and followed by number
+	//if (str.slice(0, 1) === '-' && isNumber(str.slice(1, 2))) {
+	
+	if (str.charAt(0) === '-' ) {
+		
+		if (isNumber(str.charAt(1))) {
+
+			negative = true;  
+			i = 1;
+
+		} else {
+		
+			return null;
+		}
+	}
+
+	// Dot cannot appear at the start or at the end of string
+	if (str[0] === '.' || str[str.length - 1]=== '.') {
+		return null;
+	}
+
+	for (;i < str.length; i++) {
+
+		if (isNumber(str.charAt(i)) ||  str.charAt(i) === '.' && allowDot) {
+ 
+			// Zero tolerance for dots
+			if (str.charAt(i) === '.') {
+
+				allowDot = false;
+				continue;
+			}
+
+			if (!allowDot) {
```

Comment:
Now that you use `allowDot` to "know" whether it's the integer part or the decimal part, the name definitely doesn't explain that. I would rename to `seenDot` (for which you'll need to negate its conditions).